src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/commands/dissect/fields_suggestions_after.test.ts

```diff
@@ -0,0 +1,142 @@
+      const columns7 = extractDissectColumnNames(pattern7);
+      expect(columns7).toStrictEqual(['fieldA', 'fieldB', 'fieldC']);
+
+      const pattern9 = '%%{field1}%%{field2}'; // Separator is '%'
+      const columns9 = extractDissectColumnNames(pattern9);
+      expect(columns9).toStrictEqual(['field1', 'field2']);
+
+      const pattern10 = '%{firstWord}-%{secondWord}';
+      const columns10 = extractDissectColumnNames(pattern10);
+      expect(columns10).toStrictEqual(['firstWord', 'secondWord']);
+
+      const pattern11 = '"""%{date} - %{msg} - %{ip}"""';
+      const columns11 = extractDissectColumnNames(pattern11);
+      expect(columns11).toStrictEqual(['date', 'msg', 'ip']);
+
+      const pattern12 = '"""%{ts->} %{level}"""';
+      const columns12 = extractDissectColumnNames(pattern12);
+      expect(columns12).toStrictEqual(['ts', 'level']);
+
+      const pattern13 = '"""%{+name} %{+name} %{+name} %{+name}"""';
+      const columns13 = extractDissectColumnNames(pattern13);
+      expect(columns13).toStrictEqual(['name']);
+
+      const pattern14 = '"""%{clientip} %{?ident} %{?auth} %{@timestamp}"""';
+      const columns14 = extractDissectColumnNames(pattern14);
+      expect(columns14).toStrictEqual(['clientip', 'ident', 'auth', '@timestamp']);
+    });
+  });
+
+  describe('fieldsSuggestionsAfter', () => {
+    it('should return the correct fields after the command', () => {
+      const dissectCommand = {
+        name: 'dissect',
+        args: [
+          {
+            args: [
+              {
+                name: 'field1',
+                location: {
+                  min: 39,
+                  max: 43,
+                },
+                text: 'field1',
+                incomplete: false,
+                type: 'identifier',
+              },
+            ],
+            location: {
+              min: 39,
+              max: 43,
+            },
+            text: 'field1',
+            incomplete: false,
+            parts: ['field1'],
+            quoted: false,
+            name: 'field1',
+            type: 'column',
+          },
+          {
+            name: '"%{firstWord}"',
+            location: {
+              min: 45,
+              max: 58,
+            },
+            text: '"%{firstWord}"',
+            incomplete: false,
+            type: 'literal',
+            literalType: 'keyword',
+            value: '"%{firstWord}"',
+            valueUnquoted: '%{firstWord}',
+          },
+        ],
+        location: {
+          min: 31,
+          max: 58,
+        },
+        text: 'DISSECTagent"%{firstWord}"',
+        incomplete: false,
+        type: 'command',
+      } as unknown as ESQLAstCommand;
+      const previousCommandFields = [
+        { name: 'field1', type: 'keyword' },
+        { name: 'field2', type: 'double' },
+      ] as ESQLRealField[];
+
+      const userDefinedColumns = [] as ESQLRealField[];
+
+      const result = fieldsSuggestionsAfter(
+        dissectCommand,
```

Comment:
nit: maybe *synth* API could be used here, too.

Reply:
Yes I used it to all tests https://github.com/elastic/kibana/commit/1b2e211bc8f93becf7035daae4d04e3970400a96